### PR TITLE
add milestones and reorganize

### PR DESCRIPTION
**Pull request by p-shahi**

### CODE REVIEW

File: ROADMAP.md
Comment (marten-seemann) [inline ROADMAP.md:43]:
```suggestion
**Goal**: go-libp2p optimizes its handshake latency up to the point where only increasing the speed of light would lead to further speedups. In particular, this means:
```

File: ROADMAP.md
Comment (marten-seemann) [inline ROADMAP.md:12]:
```suggestion
-   D.1 📺 Universal Browser Connectivity - **[WebRTC Browser -> Server](https://github.com/libp2p/go-libp2p/pull/1655)**
```

File: ROADMAP.md
Comment (marten-seemann) [inline ROADMAP.md:27]:
```suggestion
-   A.3 ⚡ Handshakes at the Speed of Light - **0.5 RTT data optimization** (for QUIC)
```

File: ROADMAP.md
Comment (marten-seemann) [inline ROADMAP.md:34]:
```suggestion
-   D.2 📺 Universal Browser Connectivity - **WebTransport** update to new draft versions
```

**Review by BigLep — COMMENTED** [review]
Carrying on from https://github.com/libp2p/go-libp2p/pull/1784

Thanks @p-shahi and @marten-seemann .  This is getting really close to hitting the minimum requirements needed for sharing with the rest of PL EngRes.

A couple of comments about work prioritization:
1. Can we bring Future-proof testing earlier?  The "WebRTC Browser -> Server" train is going with LBL and I want to support it.  Before picking up new initiatives though like comprehensive metrics and smart dialing, can we buckle down and get the testing effort milestone handled?  I know no one wants to do testing full time, but we'll have other ongoing "feature" work with WebRTC, Early Muxer Negotiation, and likely some WebTransport followups to blend in while we tackle the testing milestone.  
2. I feel strongest about testing moving up, but I also wonder about moving benchmarking up sooner (ahead of some of the smart dialing milestones) given the existential risk that performance perceptions can have about libp2p.  

Presentation:
1.  I personally think we should get this merged by EOD 2022-10-07 even though it isn't final.  Followup PRs can easily be created for future updates.  This improves the discoverability and conveys more strongly that this is our plan of record.  
1. Let's add a status section making clear that this is in progress, will be heavily reviewed and iterated on in 202210 based on stakeholder feedback, and have a link for where/how someone leaves comments or give input (I assume we link to a libp2p 2023 roadmap issue that we create).  
2. I know there are comments to add issues for each of the epics.  It'll be great to have those.  Those issues can have those issues link to specs, etc.
3. After merging, please update relevant places (e.g., README, libp2p/specs ROADMAP.md).

Thanks again!

**Comment by marten-seemann** [comment]
> I also wonder about moving benchmarking up sooner (ahead of some of the smart dialing milestones) given the existential risk that performance perceptions can have about libp2p.

I'd love to, but this depends on the IPDX team finishing the "Remote Runner" feature. @BigLep, do you have any estimate for when this is going to happen?

> Can we bring Future-proof testing earlier? The "WebRTC Browser -> Server" train is going with LBL and I want to support it. Before picking up new initiatives though like comprehensive metrics and smart dialing, can we buckle down and get the testing effort milestone handled?

Given the explosion of dialable addresses, Smart Dialing (B.1) is becoming more and more a prerequisite for adding new transports. That said, I think it's doable to have the RTT test done by Q4.

**Review by marten-seemann — APPROVED** [review]

**Comment by MarcoPolo** [comment]
> > I also wonder about moving benchmarking up sooner (ahead of some of the smart dialing milestones) given the existential risk that performance perceptions can have about libp2p.
> 
> I'd love to, but this depends on the IPDX team finishing the "Remote Runner" feature. @BigLep, do you have any estimate for when this is going to happen?

I think we can leverage my [remote-runners mvp](https://github.com/testground/testground/pull/1425) here just fine. I don't think we should be blocked on a perfect solution to get benchmarks out the door.

I too would love to see benchmarking moved up (why not near the front?). It will pay dividends and the sooner we set this up the sooner we can collect.

**Comment by p-shahi** [comment]
> > > I also wonder about moving benchmarking up sooner (ahead of some of the smart dialing milestones) given the existential risk that performance perceptions can have about libp2p.
> > 
> > 
> > I'd love to, but this depends on the IPDX team finishing the "Remote Runner" feature. @BigLep, do you have any estimate for when this is going to happen?
> 
> I think we can leverage my [remote-runners mvp](https://github.com/testground/testground/pull/1425) here just fine. I don't think we should be blocked on a perfect solution to get benchmarks out the door.
> 
> I too would love to see benchmarking moved up (why not near the front?). It will pay dividends and the sooner we set this up the sooner we can collect.

For the benchmarking milestone, the acceptance criteria is to introduce a new mode to Testground that allows us to spin up instances in datacenters across the world. Because this is a dependency on the IPDX team, we put its completion milestone later than we'd like.
That doesn't prevent us from using https://github.com/testground/testground/pull/1425 in the short term; but the goal of having official support for remote runners means we can't be sure it'll get done in Q4 since we don't know Testground's timeline (yet cc: @laurentsenta @BigLep )
This item is still a P0 for us though.

**Review by marten-seemann — COMMENTED** [review]
QUIC, not Quic

**Comment by MarcoPolo** [comment]
> For the benchmarking milestone, the acceptance criteria is to introduce a new mode to Testground that allows us to spin up instances in datacenters across the world. 

I don't follow why we need the ability to spin up instances. If these tests are running on every PR/commit-to-master they'll be pretty busy. We can spin them up once, define their infra as part of a Terraform file, and call it done.

> That being said. I can add to the benchmark milestone and move F.1 to Q4 (or early Q1): wdyt? F.1 - Benchmarking using nix-builders (dependency: remote machines need Nix installed) F.2 - Benchmarking using first class support for remote runners (using remote:exec) in Testground

This sound good. I think F.1 will give us >80% of the benefit of F.2. And also inform what F.2 should look like. Also worth noting that we can do F.1 _today_. We aren't blocked on anything (except time and prioritization).

**Comment by p-shahi** [comment]
Merged this into https://github.com/libp2p/go-libp2p/pull/1784 
Continue discussions there.

**Pull request merged**
